src/metric-engine/src/data_region.rs

```diff
@@ -73,7 +74,13 @@ impl DataRegion {
         let new_column_id_start = 1 + region_metadata
             .column_metadatas
             .iter()
-            .map(|c| c.column_id)
+            .filter_map(|c| {
+                if ReservedColumnId::is_reserved(c.column_id) {
+                    None
+                } else {
+                    Some(c.column_id)
+                }
+            })
```

Comment:
non-blocking: filter?
```suggestion
            .filter(|c| !ReservedColumnId::is_reserved(c.column_id))
```

Reply:
Want `ColumnId` not `ColumnMetadata`